multidict/_multidict_py.py

```diff
@@ -27,6 +27,11 @@
 else:
     from typing_extensions import Self
 
+if sys.version_info >= (3, 10):
+    from types import NotImplementedType
+else:
+    from typing import Any as NotImplementedType
```

Comment:
Try
```suggestion
    NotImplementedType = object
```
to see if it works.

Reply:
Typeshed actually types it as a subclass of Any, so it might make more sense to keep it like this. Regardless, we'll be dropping the code end of this year, so not worth spending time on.

Reply:
Dropped